### PR TITLE
Updates to Rubocop instructions

### PR DESCRIPTION
**Pull request by JonRowe**
These changes are from rspec/rspec-expectations which were reverted by rspec/rspec-expectations#1254
They should probably be applied to all repos?

### CODE REVIEW

**Comment by JonRowe** [comment]
https://github.com/rspec/rspec-core/pull/2817
https://github.com/rspec/rspec-expectations/pull/1255
https://github.com/rspec/rspec-mocks/pull/1373
https://github.com/rspec/rspec-support/pull/465

File: common_plaintext_files/BUILD_DETAIL.md.erb [hunk 1]
```diff
@@ -92,22 +92,23 @@ $ bundle exec yard doc --no-cache
...
 
-We use [Rubocop](https://github.com/bbatsov/rubocop) to enforce style conventions on the project so
-that the code has stylistic consistency throughout. Run with:
+We use [RuboCop](https://github.com/rubocop-hq/rubocop) to enforce style
+conventions on the project so that the code has stylistic consistency
+throughout. Run with:
 
 ```
-$ bundle exec rubocop lib
+$ bundle exec rubocop
```
Comment (pirj) [inline common_plaintext_files/BUILD_DETAIL.md.erb:102]: [`rspec-rails`](https://github.com/rspec/rspec-rails/pull/2272) and [`rspec-expectations`](https://github.com/rspec/rspec-expectations/pull/1104) are offence-free to my best memory, including `spec/`. `rspec-mocks` and `rspec-core` are not. Their `.rubocop.yml` files are different, and only `lib/` is offence-free.
Unfortunately, it is not consistent.

Comment (JonRowe) [reply common_plaintext_files/BUILD_DETAIL.md.erb]: Ok, we'll just stick to `lib` for all repos.

**Review by pirj — APPROVED** [review]

**Review by benoittgt — APPROVED** [review]
LGTM. Merry Christmas. 🎅

**Comment by JonRowe** [comment]
https://github.com/rspec/rspec-core/pull/2818
https://github.com/rspec/rspec-expectations/pull/1256
https://github.com/rspec/rspec-mocks/pull/1374
https://github.com/rspec/rspec-support/pull/466
https://github.com/rspec/rspec-rails/pull/2429/files (with parts manually reverted)

**Pull request merged**